### ISSUE TITLE
Pi-hole as DNS Server

### ISSUE THREAD

**Issue by MajorTomDE** [WeeJeWel/wg-easy#201]
Hey,

I switched from PiVPN to wg-easy.
PiVPN has the option "Found Pi-hole. Do you want to use Pi-hole as DNS?"

How can I also use Pi-hole as DNS Server in wg-easy?

I tried to to set my Pi-hole IP as WG_DEFAULT_DNS instead of 1.1.1.1, but no function. 

Thank you
Thomas

**Comment by Teraskull** [comment]
Add the IP address of the pi in the client `.conf`:
```ini
DNS = 192.168.X.X
```

**Comment by MajorTomDE** [comment]
Hey,
I found the problem. It seems it depends of the networks of the docker volumes

My Docker host has IP 192.168.178.2
wg-easy Container is set as Bridge Network
pi-hole Container is set as macvlan with IP 192.168.178.201

With this configuration I have no access to pi-hole DNS 192.168.178.201/macvlan

So, I configured wg-easy Container also as macvlan with IP 192.168.178.202
With this I have access to pi-hole DNS 192.168.178.201/macvlan
But now I have no access to my webserver on host 192.168.178.2

And now the tricky question:
How can I configure wg-easy that I will have access to
192.168.178.201/macvlan and host 192.168.178.2?
(In best case wg-easy has its own IP with macvlan)

Thank you
Tom

**Comment by CyberiumShadow** [comment]
> Hey, I found the problem. It seems it depends of the networks of the docker volumes
> 
> My Docker host has IP 192.168.178.2 wg-easy Container is set as Bridge Network pi-hole Container is set as macvlan with IP 192.168.178.201
> 
> With this configuration I have no access to pi-hole DNS 192.168.178.201/macvlan
> 
> So, I configured wg-easy Container also as macvlan with IP 192.168.178.202 With this I have access to pi-hole DNS 192.168.178.201/macvlan But now I have no access to my webserver on host 192.168.178.2
> 
> And now the tricky question: How can I configure wg-easy that I will have access to 192.168.178.201/macvlan and host 192.168.178.2? (In best case wg-easy has its own IP with macvlan)
> 
> Thank you Tom

You might need to look into setting up a macvlan-shim as well if your web server is on the same physical host as the pihole and wireguard macvlans

**Comment by shizunge** [comment]
I added 
```
iptables -t nat -A PREROUTING -p udp -d 10.8.0.1 --dport 53 -j DNAT --to-destination ${NETWORK_DNS_IP}:53
```
to my post up script, where NETWORK_DNS_IP is my DNS IP. If your DNS and wg-easy in the same docker network, it could be a private IP within the docker network.

And set  `WG_DEFAULT_DNS=10.8.0.1`, where 10.8.0.1 is the gateway/server wireguard IP address.

**Comment by rmayobre** [comment]
Try this out: https://github.com/rmayobre/easy-pwn

**Comment by ShlomiD83** [comment]
> > Hey, I found the problem. It seems it depends of the networks of the docker volumes
> > My Docker host has IP 192.168.178.2 wg-easy Container is set as Bridge Network pi-hole Container is set as macvlan with IP 192.168.178.201
> > With this configuration I have no access to pi-hole DNS 192.168.178.201/macvlan
> > So, I configured wg-easy Container also as macvlan with IP 192.168.178.202 With this I have access to pi-hole DNS 192.168.178.201/macvlan But now I have no access to my webserver on host 192.168.178.2
> > And now the tricky question: How can I configure wg-easy that I will have access to 192.168.178.201/macvlan and host 192.168.178.2? (In best case wg-easy has its own IP with macvlan)
> > Thank you Tom
> 
> You might need to look into setting up a macvlan-shim as well if your web server is on the same physical host as the pihole and wireguard macvlans

Thats exactly what worked for me.
Using a macvlan-shim I'm able to use AdGuard Home as my dns when connecting to Wireguard.

**Comment by WeeJeWel** [comment]
See https://github.com/WeeJeWel/wg-easy/wiki/Using-WireGuard-Easy-with-Pi-Hole

**Comment by bluepuma77** [comment]
Hi @WeeJeWel, thanks for supplying [extra documentation](https://github.com/WeeJeWel/wg-easy/wiki/Using-WireGuard-Easy-with-Pi-Hole) for this, awesome!

May I ask if you have tested it yourself? I tried to deploy "WireGuard Easy with Pi Hole" into a small Ubuntu VM and it wouldn't work because port 53 (DNS) was already in use. So I disabled the local Ubuntu DNS service
```
sudo systemctl disable systemd-resolved.service
sudo systemctl stop systemd-resolved
```
but now the Pi-Hole blocklist update fails because of unreachable DNS ;-)

Pi-Hole GUI: `http://hostname:5353/admin/` -> Tools -> Update Gravity

<img width="860" alt="PiHole-Update-Gravity-DNS" src="https://user-images.githubusercontent.com/5221187/177012732-351c3065-7b84-4f05-ae20-78829b9c4d08.png">

(I admit I didn't dare to open yet another new ticket for Pi-Hole)
